src/worldfactory.cpp

```diff
@@ -1414,6 +1414,33 @@ void load_world_option( JsonObject &jo )
     }
 }
 
+void load_external_option( JsonObject &jo )
+{
+    auto name = jo.get_string( "name" );
+    auto stype = jo.get_string( "stype" );
+    options_manager &opts = get_options();
+    if( !opts.has_option( name ) ) {
+        auto sinfo = jo.get_string( "info" );
+        opts.add_external( name, "world_default", stype, sinfo, sinfo );
+    }
+    options_manager::cOpt &opt = opts.get_option( name );
+    if( stype == "float" ) {
+        opt.setValue( ( float ) jo.get_float( "value" ) );
```

Comment:
`get_float` should already return a float, why the cast?

Reply:
Better to use `static_cast<float>()` if it's necessary

Reply:
Nope it is return double. Because logic.

Reply:
`double` is pretty standard, the `float` type is actually not very precise.  The correct "fix" would probably be to make options use doubles instead of floats

Reply:
It may cause unexpected behavior and force to do a lot more things that is not target of this PR. So let's move it out of this PR.